Enterprise-CMCS/cmcs-eregulations
# EREGCSC-2862 -- Update sidebar for subjects pages

Resolves [EREGCSC-2862](https://jiraent.cms.gov/browse/EREGCSC-2682)

**Description**

Updates left sidebar on `/subjects` page to match [Figma design](https://www.figma.com/design/LfXWcXJI49UaxMfFTsQHZO/Browse-without-Search?node-id=1-18968&node-type=frame&t=WZzWcJZqfFDePuVR-0).

**This pull request changes:**

- Uses `position: sticky` to fix left sidebar in place so it does not scroll with the rest of the page
   - Left sidebar can now scroll independently
- Subject Selector in left sidebar is restyled to match Figma design
- Subject Selector behavior changes:
   - Selected subject is displayed underneath text input in left sidebar
   - Selected subject and text input stick to the top of the page when left sidebar is scrolled -- the are always visible
   - Selected subject is removed from subject list
   - When subject is unselected, it is added back to subject list

**Steps to manually verify this change:**

1. Green check marks
2. [Visit experimental deployment](https://i52gwakusd.execute-api.us-east-1.amazonaws.com/dev1487)
3. Compare experimental deployment to [Figma comp](https://www.figma.com/design/LfXWcXJI49UaxMfFTsQHZO/Browse-without-Search?node-id=1-18971&node-type=frame&m=dev) to see if they match
4. Select a subject and make sure it is removed from the subject list and re-added to the list when unselected
5. Arrive at page with a selected subject and ensure everything works as expected
6. Visit `/search` and use the Subjects dropdown and ensure there have not been any changes or regressions
